src/components/Select/Select.tsx

```diff
@@ -23,34 +24,9 @@ export type SelectProps = Omit<React.HTMLAttributes<HTMLDivElement>, 'value' | '
 
 enum Mode {
   Resting,
-  Open,
+  Open
 }
 
-type State = {
-  value: string;
-  focused: string | null;
-  mode: Mode;
-};
-
-type Action =
-  | { type: 'OPEN' }
-  | { type: 'CLOSE' }
-  | { type: 'SELECT'; payload: { value: string } }
-  | { type: 'FOCUS'; payload: { value: string } };
-
-const reducer = (state: State, action: Action) => {
-  switch (action.type) {
-    case 'OPEN':
-      return { ...state, mode: Mode.Open };
-    case 'CLOSE':
-      return { ...state, mode: Mode.Resting };
-    case 'SELECT':
-      return { ...state, value: action.payload.value, focused: null, mode: Mode.Resting };
-    case 'FOCUS':
-      return { ...state, focused: action.payload.value };
-  }
-};
-
```

Comment:
wanted to get rid of those before / feeling like this is overthinked, this's a simple component

Reply:
I wouldn't get rid of `useReducer`, I think it's more readable than having multiple `useState`s.